File: content/snippets/typescript/exercises/advanced/n_queens_problem.ts
const N = 8
const cols = new Array<number>(N).fill(0)
function ok(r:number,c:number){ for(let i=0;i<r;i++) if(cols[i]===c || Math.abs(cols[i]-c)===r-i) return false; return true }
function solve(r=0): boolean{ if(r===N){ console.log(cols); return true } for(let c=0;c<N;c++){ if(ok(r,c)){ cols[r]=c; if(solve(r+1)) return true } } return false }
solve(); export {}

File: content/snippets/cpp/exercises/advanced/n_queens_problem.cpp
#include <bits/stdc++.h>
using namespace std;
int N=8; vector<int> col;
bool ok(int r,int c){ for(int i=0;i<r;i++) if(col[i]==c||abs(col[i]-c)==r-i) return false; return true; }
void solve(int r){ if(r==N){ for(int i=0;i<N;i++) cout<<col[i]<<" "; cout<<"\n"; exit(0);} for(int c=0;c<N;c++) if(ok(r,c)){ col[r]=c; solve(r+1); }}
int main(){ col.assign(N,0); solve(0); }
